geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingFunction.java

```diff
@@ -75,12 +80,29 @@ public CliFunctionResult executeFunction(FunctionContext<Object[]> context)
    * and the given async-event-queue as one of its queues.
    */
   private void alterRegion(Region<?, ?> region, String queueName, boolean synchronous) {
-    region.getAttributesMutator().setCacheLoader(new JdbcLoader());
-    if (synchronous) {
-      region.getAttributesMutator().setCacheWriter(new JdbcWriter());
+    boolean isProxy = isProxyRegion(region);
+    if (isProxy) {
+      if (!synchronous) {
+        region.getAttributesMutator().addAsyncEventQueueId(queueName);
+      }
+    } else {
+      region.getAttributesMutator().setCacheLoader(new JdbcLoader());
+      if (synchronous) {
+        region.getAttributesMutator().setCacheWriter(new JdbcWriter());
+      } else {
+        region.getAttributesMutator().addAsyncEventQueueId(queueName);
+      }
+    }
+  }
+
+  private boolean isProxyRegion(Region<?, ?> region) {
+    boolean isProxy = false;
+    if (region instanceof PartitionedRegion) {
```

Comment:
I think this code would be cleaner if you just had it work off the RegionAttributes that you can get from the passed in Region. That way you would need need instanceof checks and casting to internal classes (PartitonedRegion and LocalRegion). You can just ask the attributes if the DataPolicy is empty (!withStorage()). If not then check if the data policy is partitioned (withPartitioning()). If so call getParititionAttributes().getLocalMaxMemory() and see if it is zero.